controllers/containerimagebuild_controller.go

```diff
@@ -118,6 +118,7 @@ func (r *ContainerImageBuildReconciler) Reconcile(req ctrl.Request) (ctrl.Result
 
 		build.Status.SetState(forgev1alpha1.Failed)
 		build.Status.ErrorMessage = err.Error()
+		build.Status.BuildCompletedAt = &metav1.Time{Time: time.Now()}
```

Comment:
I guess it makes sense to set the completed at datetime after a failure. Good catch.